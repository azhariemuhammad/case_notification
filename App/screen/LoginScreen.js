import React, {useState} from 'react'
import {View, Text, TextInput, TouchableOpacity, StyleSheet} from 'react-native'
import {baseService} from "../services";
import {getToken, setToken} from "../utils/storeToken";



const LoginScreen = (props) => {
    const [email, setEmail] = useState("")
    const [phone, setPhone] = useState("082248467118")
    const [password, setPassword] = useState("12345qwe")

    const handleChangePhoneOrEmail = (phone) => {
        setPhone(phone)
    };

    const handlePassword = (pass) => {
        setPassword(pass)
    }

    const handleSubmit = async () => {
        // TODO:
        // handle login by email
        const dataUser = {
            phone: phone,
            password: password
        }
        await baseService().login(dataUser).then(res => {
            setToken(res.data.token)
            props.navigation.navigate(res.data.token ? 'App' : 'Auth');

        }).catch(e => {
            console.log('Somenthing went wrong', e)
        })
    }

    return (
        <View>
            <TextInput
                underlineColorAndroid="transparent"
                placeholder="Email/Telepon"
                placeholderTextColor="black"
                autoCapitalize="none"
                value={phone || ""}
                onChangeText={handleChangePhoneOrEmail}
            />
            <TextInput
                underlineColorAndroid="transparent"
                placeholder="Password"
                placeholderTextColor="black"
                autoCapitalize="none"
                secureTextEntry={true}
                value={password || ""}
                onChangeText={handlePassword}
            />

            <TouchableOpacity
                onPress={() => handleSubmit()}
                style={styles.button}
            >
                <Text> Masuk </Text>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    button: {
        alignItems: 'center',
        backgroundColor: '#DDDDDD',
        padding: 10
    }
})

export default LoginScreen
