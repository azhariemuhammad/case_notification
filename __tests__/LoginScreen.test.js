import 'react-native';
import React from 'react';
import {create} from 'react-test-renderer';

import LoginScreen from "../App/screen/LoginScreen";


it("should render LoginScreen and compare with snapshot", async () => {
    const tree = create(<LoginScreen/>).toJSON()
    expect(tree).toMatchSnapshot()
})
