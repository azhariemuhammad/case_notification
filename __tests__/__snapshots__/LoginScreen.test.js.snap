// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginScreen and compare with snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#226597",
        "height": 284,
      }
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "height": 70,
            "marginBottom": 16,
            "marginTop": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          Telepon
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="08XXXXX"
          placeholderTextColor="#bfc6ea"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#ECF0F3",
              "marginTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "height": 70,
            "marginBottom": 16,
            "marginTop": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          Sandi
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Sandi"
          placeholderTextColor="#bfc6ea"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#ECF0F3",
              "marginTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "marginTop": 30,
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#007aff",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
             Masuk 
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
